tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh

```diff
@@ -373,6 +376,65 @@ install_nvidia_dcgm() {
 		datacenter-gpu-manager-exporter
 }
 
+cleanup_rootfs() {
+	echo "chroot: Cleanup NVIDIA GPU rootfs"
+
+	apt-mark hold libstdc++6 libzstd1 libgnutls30t64 pciutils
+
+	if [[ -n "${driver_version}" ]]; then
+		apt-mark hold libnvidia-cfg1-"${driver_version}"-server \
+			nvidia-utils-"${driver_version}"-server         \
+			nvidia-kernel-common-"${driver_version}"-server \
+			nvidia-imex-"${driver_version}"                 \
+			nvidia-compute-utils-"${driver_version}"-server \
+			libnvidia-compute-"${driver_version}"-server    \
+			libnvidia-gl-"${driver_version}"-server         \
+			libnvidia-extra-"${driver_version}"-server      \
+			libnvidia-decode-"${driver_version}"-server     \
+			libnvidia-fbc1-"${driver_version}"-server       \
+			libnvidia-encode-"${driver_version}"-server     \
+			libnvidia-nscq-"${driver_version}"              \
+			linuxptp libnftnl11
+	fi
+
+	kernel_headers=$(dpkg --get-selections | cut -f1 | grep linux-headers)
+	linux_images=$(dpkg --get-selections | cut -f1 | grep linux-image)
+	for i in ${kernel_headers} ${linux_images}; do
+		apt purge -yqq "${i}"
+	done
+
+	apt purge -yqq jq make gcc xz-utils linux-libc-dev
+
+	if [[ -n "${driver_version}" ]]; then
+		apt purge -yqq nvidia-headless-no-dkms-"${driver_version}"-server"${driver_type}" \
+			nvidia-kernel-source-"${driver_version}"-server"${driver_type}" -yqq
+	fi
+
+	apt autoremove -yqq
+
+	apt clean
+	apt autoclean
+
+	for modules_version in /lib/modules/*; do
+		ln -sf "${modules_version}" /lib/modules/"$(uname -r)"
+		touch  "${modules_version}"/modules.order
+		touch  "${modules_version}"/modules.builtin
+		depmod -a
+	done
+
```

Comment:
We need to leave the next two lines for the devkit so apt works properly do not remove sources.list and the /var/lib/apt stuff.